packages/kernel/kernel.spec

```diff
@@ -84,35 +90,83 @@ find %{buildroot}%{_cross_prefix} \
    \( -name .install -o -name .check -o \
       -name ..install.cmd -o -name ..check.cmd \) -delete
 
-# files for external module compilation
+# For out-of-tree kmod builds, we need to support the following targets:
+#   make scripts -> make prepare -> make modules
+#
+# This requires enough of the kernel tree to build host programs under the
+# "scripts" and "tools" directories.
+
+# Any existing ELF objects will not work properly if we're cross-compiling for
+# a different architecture, so get rid of them to avoid confusing errors.
+find arch scripts tools -type f -executable \
+  -exec sh -c "head -c4 {} | grep -q ELF && rm {}" \;
+
+# We don't need to include these files.
+find -type f \( -name \*.cmd -o -name \*.gitignore \) -delete
+
+# Avoid an OpenSSL dependency by stubbing out options for module signing and
+# trusted keyrings, so `sign-file` and `extract-cert` won't be built. External
+# kernel modules do not have access to the keys they would need to make use of
+# these tools.
+sed -i \
+  -e 's,$(CONFIG_MODULE_SIG_FORMAT),n,g' \
+  -e 's,$(CONFIG_SYSTEM_TRUSTED_KEYRING),n,g' \
+  scripts/Makefile
+
 (
-  find * -name Kbuild\* -type f -print  \
-    -o -name Kconfig\* -type f -print \
-    -o -name Makefile\* -type f -print \
-    -o -name module.lds -type f -print \
-    -o -name Platform -type f -print
-  find arch/*/include/ include/ -type f -o -type l
-  find scripts/ -executable -type f
-  find scripts/ ! \( -name Makefile\* -o -name Kbuild\* \) -type f
+  find * \
```

Comment:
I'm not sure how to do better than the comment that says that this is what we need for the three targets (scripts, prepare, modules). The set of files is subject to change over time since the authoritative source is the kernel tree for a specific version.

"Keep adding things until those targets work" and "stop adding things once they do" is unsatisfying advice, but it pretty accurately reflects the workflow.

Pointing to a different packaging effort can be misleading since the set of files changes over time and can be influenced by the kernel configuration. We also need more of the sources than others might because we can't include the host tools.